enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,205 @@
+The Numa-Resources-Operator (NROP) is responsible for deploying and enabling
+TAS capabilities on the cluster.
+This proposal aims to address the gaps identified
+during the installation of NROP on the HyperShift (HCP) platform.
+
+## Motivation
+
+See the HCP (HyperShift) project goals.
+These goals should not come at the cost of giving up any existing OCP (OpenShift) features.
+On standard OCP it is possible to deploy NROP (Numa-Resources-Operator) as a day-2 operator 
+to optimize scheduling decisions based on the node's hardware layout.
+It's a desire to have the same functionality on HyperShift as well.
+
+
+### User Stories
+
+* As a cluster service consumer, I want to be able to use TAS capabilities on HyperShift's hosted clusters in a similar fashion to standard OCP clusters.
+
+* As an administrator, I want to be able to install NROP on HyperShift's hosted clusters in a similar fashion to standard OCP clusters.
+
+### Goals
+
+Allow a day-2 NROP deployment on a hosted cluster 
+to support TAS (Topology-Aware-Scheduling) over HyperShift platform.
+
+### Non-Goals
+
+- N/A
+
+## Proposal
+
+NROP is a day-2, optional operator that users can install on demand.
+To make the procedure as similar as possible to standard OCP clusters,
+it is suggested to install the operator and its managed components on the hosted-cluster.
+The following paragraphs detail the identified gaps and suggest resolution for each. 
+
+### Architectural Overview
+
+TAS solution comprises three main components:
+
+Resource-Topology-Exporter (RTE) - Responsible for collecting node information (mostly NUMA resources allocations) 
+and advertises it as a CR object named NRT (Node-Resource-Topology)
+
+Secondary-Scheduler - A complementary scheduler to the default scheduler
+which enhances the cluster and responsible for
+scheduling the pod in the most suitable node according to the data exposed by the NRT object.
+
+NROP - The operator itself which is responsible for the deployment of RTE and the Scheduler.
+RTE and the scheduler are configurable via the NumaResourceOperator and NumaResourceScheduler CRs respectively.
+
+### Expose KubeletConfig to Hosted-Clusters
+
+NROP needs to learn the topology-manager (TM) scope and policy of the nodes on which TAS is running
+and feed RTE with that information.
+RTE advertises the TM configuration in the NRT object which is used by the scheduler  
+to perform scheduling logic as close as possible as the topology manager.
+For that, NROP needs access to the KubeletConfig object which holds the information about the TM scope and policy.
+
+Since NROP is running on the hosted-cluster, it doesn’t have access to the KubeletConfig object.
+On HyperShift, the Hosted-Cluster-Config-Operator (HCCO) is responsible for rendering the KubeletConfig objects
+and applying their configuration on the node.
+It’s suggested that when HCCO renders KubeletConfig, it adds a ConfigMap that encapsulates the KubeletConfig data
```

Comment:
I would like to see a little bit more details here:
* How will the HCCO get access to the KubeletConfigs? (They are available to the nodepool controller, but not necessarily in the control plane namespace).
* Where will the kubelet config CMs live in the hosted cluster? (Maybe in `openshift-config-managed`?) 
* What will the CMs be named? How will they be labeled? 
* What is expected of nodepools for which a kubeletconfig has not been specified?
* Will the HCCO delete the CM of nodepools that are removed?

Reply:
AFAIU HCCO is the only controller who's responsible to creates objects on the hosted cluster, but I couldn't figure out how it works, I will invest more time trying to find a way.
We can mirror KubeletConfig CM to the control plane namespace with the help of the nodePool controller.